library/alloc/src/rc.rs

```diff
@@ -2015,12 +2485,14 @@ impl<T> From<Vec<T>> for Rc<[T]> {
     /// assert_eq!(vec![1, 2, 3], *shared);
     /// ```
     #[inline]
-    fn from(mut v: Vec<T>) -> Rc<[T]> {
+    fn from(v: Vec<T, A>) -> Rc<[T], A> {
         unsafe {
-            let rc = Rc::copy_from_slice(&v);
-            // Allow the Vec to free its memory, but not destroy its contents
-            v.set_len(0);
-            rc
+            let (vec_ptr, len, _, alloc) = v.into_raw_parts_with_alloc();
+
+            let rc_ptr = Self::allocate_for_slice_in(len, &alloc);
+            ptr::copy_nonoverlapping(vec_ptr, &mut (*rc_ptr).value as *mut [T] as *mut T, len);
+
+            Self::from_ptr_in(rc_ptr, alloc)
```

Comment:
```suggestion
            let rc_ptr = Self::copy_from_slice_in(&v, v.allocator());
            // Allow the Vec to free its memory, but not destroy its contents
            v.set_len(0);
            rc
```

Could refactor the rest to `copy_from_slice_in` to parallel `copy_from_slice`

Reply:
Ah, good point. Maybe you could split it into parts then re-form the `RawVec` using `&A` (since `impl Allocator for &A` exists) and let that deallocate itself?

Reply:
D'oh, that exists, no `set_len` call even needed

```rust
let (vec_ptr, len, cap, alloc) = v.into_raw_parts_with_alloc();
let _ = Vec::from_raw_parts_in(vec_ptr, 0, cap, &alloc);
```